source/jormungandr/jormungandr/georef.py

```diff
@@ -158,21 +158,30 @@ def get_crow_fly(
             logger.error("feed publisher not empty: expect performance regression!")
         return res.places_nearby
 
-    def get_stop_points_for_stop_area(self, uri, request_id):
+    def get_stop_points_uris_for_stop_area(self, stop_area_uri, request_id):
         req = request_pb2.Request()
         req.requested_api = type_pb2.PTREFERENTIAL
         req.ptref.requested_type = type_pb2.STOP_POINT
         req.ptref.count = 100
         req.ptref.start_page = 0
         req.ptref.depth = 0
-        req.ptref.filter = 'stop_area.uri = {uri}'.format(uri=uri)
+        req.ptref.filter = 'stop_area.uri = {uri}'.format(uri=stop_area_uri)
+        return self.instance.send_and_receive(req, request_id=request_id)
 
-        result = self.instance.send_and_receive(req, request_id=request_id)
-        if not result.stop_points:
-            logging.getLogger(__name__).info(
-                'PtRef, Unable to find stop_point with filter {}'.format(req.ptref.filter)
-            )
-        return result.stop_points
+    def get_stop_points_for_stop_area(self, uri, request_id):
+        @memory_cache.memoize(
+            current_app.config[str('MEMORY_CACHE_CONFIGURATION')].get(str('TIMEOUT_PTOBJECTS'), 30)
+        )
+        @cache.memoize(current_app.config[str('CACHE_CONFIGURATION')].get(str('TIMEOUT_PTOBJECTS'), 300))
+        def inner(stop_area_uri, instance_publication_date):
+            result = self.get_stop_points_uris_for_stop_area(stop_area_uri, request_id)
```

Comment:
you can capture the exception "Deadsocket" here and log it then return an empty list, so even though kraken doesn't answer, the request can carry on